feat: inherit stdin when running starship

fa-mi-lu-si opened the pull request:
Closes #15

Allows starship to auto detect the terminal width by inheriting stdin from yazi.

Rolv-Apneseth commented:
Would you mind sharing your starship theme that you were using for the screenshot posted on the issue, so that I can confirm this works to resolve the problem? Also,  not sure I understood, so just to clarify, this is so that prompts which align to the end of the terminal show up correctly?

fa-mi-lu-si commented:
Here is the starship theme, it is the [fill module](https://starship.rs/config/#fill) that is responsible for aligning modules to the right-hand side.

```toml
format = "${env_var.YAZI_LEVEL}$directory$python$rust$nix_shell$fill${custom.git_host}$git_branch$git_commit$git_state$git_status$package$line_break$username$hostname$character"
right_format = "$sudo$jobs$battery"
[[battery.display]]
charging_symbol = "󰂄 "
discharging_symbol = "󰂃 "
threshold = 30

[custom.git_host]
command = "IFS=\"/\"\nGIT_REMOTE=$(command git ls-remote --get-url 2> /dev/null)\nread -ra GIT_HOST <<< \"$GIT_REMOTE\"\n\nif [[ \"$GIT_REMOTE\" =~ \"github\" ]]; then\n    GIT_REMOTE_SYMBOL=\" \"\nelif [[ \"$GIT_REMOTE\" =~ \"gitlab\" ]]; then\n    GIT_REMOTE_SYMBOL=\" \"\nelif [[ \"$GIT_REMOTE\" =~ \"bitbucket\" ]]; then\n    GIT_REMOTE_SYMBOL=\" \"\nelif [[ \"$GIT_REMOTE\" =~ \"git\" ]]; then\n    GIT_REMOTE_SYMBOL=\" \"\nelse\n    GIT_REMOTE_SYMBOL=\" \"\nfi\n\necho \"$GIT_REMOTE_SYMBOL\"\nIFS=\"\"\n"
description = "Display symbol for remote Git server"
directories = [".git"]
format = "[$output]($style) "
shell = ["sh", "--norc", "--noprofile"]
style = "white bold"
when = "git rev-parse --is-inside-work-tree 2> /dev/null"

[custom.git_last_commit]
command = "git log --pretty=format:'%cr' -1\n"
description = "Show time since last commit"
directories = [".git"]
format = "[ $output]($style) "
shell = ["sh", "--norc", "--noprofile"]
style = "yellow bold"
when = "git rev-parse --is-inside-work-tree 2> /dev/null"

[directory]
before_repo_root_style = "bold cyan"
format = "[$path]($style)[$read_only]($read_only_style) "
read_only = " "
repo_root_format = "[ $repo_root]($repo_root_style)[$path]($style)[$read_only]($read_only_style) "
repo_root_style = "bold cyan"
style = "bold cyan"
truncation_length = 3
truncation_symbol = "…/"

[directory.substitutions]
"~/.config" = ".󰒓 "
"~/.dotfiles" = ". "
"~/Desktop" = "󰇄 "
"~/Documents" = "󰈙 "
"~/Downloads" = " "
"~/Music" = "󰝚 "
"~/Pictures" = " "
"~/Projects" = " "
"~/Vault" = "󰠮 "
"~/Videos" = " "

[env_var.YAZI_LEVEL]
description = "Display for yazi shell"
format = "[󰇥 ]($style)"
style = "yellow bold"

[fill]
symbol = " "

[git_branch]
format = "[$symbol $branch]($style) "
symbol = ""

[git_state]
format = "([$state( $progress_current/$progress_total)]($style)) "

[git_status]
ahead = "⇡$count "
behind = "[⇣$count](red)"
conflicted = "🏳"
deleted = "󰗨 $count "
diverged = "⇕⇡$ahead_count [⇣$behind_count](red) "
format = "[($conflicted$untracked$modified$staged$renamed$deleted)($ahead_behind$stashed)]($style)"
modified = " $count "
renamed = "󰾵 "
staged = "[++$count](green) "
stashed = " $count "
style = "red"
untracked = "[ $count](sapphire) "
up_to_date = ""

[line_break]
disabled = false

[nix_shell]
format = "[$symbol$state]($style) "
heuristic = false
impure_msg = "~ "
pure_msg = "󰡱 "
symbol = " "

[python]
format = "[$symbol$virtualenv]($style)"
symbol = " "

[rust]
format = "via [$symbol]($style)"

[sudo]
disabled = false
format = "[$symbol]($style)"
symbol = "󱑷 "

```

Rolv-Apneseth commented:
Seems to work well enough yeah, and nice theme by the way.

Did you make the PR a draft for a specific reason?

fa-mi-lu-si commented:
I just wanted to wait for some feedback in case you had any suggestions.

Rolv-Apneseth commented:
As far as I can tell it works as intended, if there's an issue with the approach I'm sure someone else will open a new issue and I'll try one of your other suggested solutions.

Thank you for the PR
